sdk/java-sdk-protobuf/src/main/scala/kalix/javasdk/impl/action/ActionsImpl.scala

```diff
@@ -122,7 +123,7 @@ private[javasdk] final class ActionsImpl(
   import _system.dispatcher
   implicit val system: ActorSystem = _system
   val telemetry = Telemetry(system)
-  val telemetries: Map[String, Instrumentation] = services.values.map { s =>
+  val telemetries: Map[String, Future[Instrumentation]] = services.values.map { s =>
```

Comment:
We talked about a fast-future to avoid mapping/scheduling for each request, but another idea, maybe we touched on that before, would be to make `telemetries` a lazy val so it isn't initialised on first request (then we know the value is set, no request can come in before discovery completed), that way we could use an atomic in the extension and avoid futures completely.

Reply:
I like the idea. The downside I see is that we get away from the types and we may break it in the future without noticing. I will add some comments but I have the feeling that we are moving from an explicit complexity (the Futures an so on) to an implicit one that the type system can't check and it's on us to keep it sound.

Reply:
It can't never pass through request to the SDK without discovering. Totally right there. I was thinking on one of us instantiating a trace instrumentation without using lazy or using it before hand. That would break it. 
As I said I like the idea, the code is more terse with your proposal. 
I changed it already. Please, have a look. I used `@volatile` and `lazy`